### PR TITLE
[SC-773] Apply config and logger.

### PR DESCRIPTION
**Pull request by hyerim-kim**
SC-773 Apply config and logger.
- loguru
- pydantic

SC-773 Edit github workflows.

### CODE REVIEW

File: .github/workflows/unit_test.yml
Comment (yakkle) [inline .github/workflows/unit_test.yml:1]: add python version(`3.8~3.11`) for tests.

File: didsdk/__about__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+__version__ = "0.0.1"
```
Comment (yakkle) [inline didsdk/__about__.py:1]: Is `0.0.1` real version?

File: didsdk/config/config_logger.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    # change handler for default uvicorn logger
+    default_handler = InterceptUvicornHandler()
+    access_handler = InterceptUvicornHandler()
+    access_handler.setFormatter(access_formatter)
+
+    logging.getLogger("uvicorn.access").handlers = [access_handler]
+
+    for _log in ["uvicorn", "uvicorn.error", "fastapi"]:
+        _logger = logging.getLogger(_log)
+        _logger.handlers = [default_handler]
```
Comment (yakkle) [inline didsdk/config/config_logger.py:92]: for Uvicorn and fastapi.

File: didsdk/config/config_logger.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+    # Intercept logger handler using by yirgachefe
+    _logger = logging.getLogger("main")
+    _logger.handlers = [BaseInterceptHandler()]
```
Comment (yakkle) [inline didsdk/config/config_logger.py:96]: for using yirgachefe.

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+keywords = []
+authors = [
+  { name = "PARAMETA Corp", email = "t_pds@parametacorp.com" },
+]
+classifiers = [
+  "Development Status :: 4 - Beta",
+  "Programming Language :: Python",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: Implementation :: CPython",
+  "Programming Language :: Python :: Implementation :: PyPy",
```
Comment (yakkle) [inline pyproject.toml:20]: add python versions.

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+[tool.coverage.report]
+exclude_lines = [
+  "no cov",
+  "if __name__ == .__main__.:",
+  "if TYPE_CHECKING:",
+]
+
+[tool.black]
+line-length = 120
+target-version = ["py311"]
```
Comment (yakkle) [inline pyproject.toml:89]: only for `py311`?

File: pyproject.toml [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+  "pytest",
+  "pytest-cov",
+]
+
+[tool.hatch.envs.default.scripts]
+cov = "pytest --cov-report=term-missing --cov-config=pyproject.toml --cov=app {args}"
+no-cov = "cov --no-cov {args}"
+
+[[tool.hatch.envs.test.matrix]]
+python = ["311"]
```
Comment (yakkle) [inline pyproject.toml:66]: Maybe unused, but this python must be set minimum version or all support versions.

File: didsdk/__about__.py (hunk 1 above)
Comment (hyerim-kim) [reply didsdk/__about__.py:1]: It's ........ never released, but it might be upgraded I guess.

File: didsdk/config.py [hunk 7]
```diff
@@ -0,0 +1,14 @@
+from pydantic import BaseSettings
+
+
+class DidSettings(BaseSettings):
+    PROJECT_NAME: str = "did-sdk-python"
+    TX_RETRY_COUNT: int = 5
+    TX_SLEEP_TIME: int = 2
```
Comment (yakkle) [inline didsdk/config.py:7]: How to set these options in external project?

File: didsdk/__about__.py [hunk 8]
```diff
@@ -0,0 +1 @@
+__version__ = "0.0.2"
```
Comment (yakkle) [inline didsdk/__about__.py:1]: https://semver.org/

I think that this PR is not just patch.
Why don't you bump up major version?

File: didsdk/config.py [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+    LOG_ENABLE_DID_LOGGER: bool = False
```
Comment (yakkle) [inline didsdk/config.py:8]: How to enable DID logger in exteral project(e.g. parameta-w)?

Comment (hyerim-kim) [reply didsdk/config.py:8]: You can add some configurations for did-sdk in a parameta-w's `.env` file.

File: didsdk/config.py (hunk 7 above)
Comment (hyerim-kim) [reply didsdk/config.py:7]: You can add some configurations for did-sdk in an external project(e.g. parameta-w)'s .env file.

File: didsdk/config.py (hunk 9 above)
Comment (hyerim-kim) [reply didsdk/config.py:8]: image

Comment (yakkle) [reply didsdk/config.py:8]: 😮

File: didsdk/config.py (hunk 7 above)
Comment (yakkle) [reply didsdk/config.py:7]: How to set 500 ms as `SLEEP_TIME`?

File: didsdk/__about__.py (hunk 8 above)
Comment (hyerim-kim) [reply didsdk/__about__.py]: You're right. But when I read the link that you gave, I think it's not a case to increase the major verion. Instead, applied increasing minor version.

File: didsdk/config.py (hunk 7 above)
Comment (hyerim-kim) [reply didsdk/config.py]: Ummm.. test cases are looking at a public network and it needs time more time. if the default time is set as 500ms, the retry count has to be 23 at least(by test).

Comment (yakkle) [reply didsdk/config.py]: What I mean is that SLEEP_TIME does not support time in milliseconds.
If W want to set 800 ms, 1500 ms, 2300 ms, etc. for waiting response?

Comment (hyerim-kim) [reply didsdk/config.py]: ummm? I tested with sleep time as 0.5 and it worked.

File: didsdk/config.py [hunk 10]
```diff
@@ -0,0 +1,24 @@
+from pydantic import BaseSettings, validator
+
+
+class DidSettings(BaseSettings):
+    PROJECT_NAME: str = "did-sdk-python"
+    TX_RETRY_COUNT: int = 5
+    # Second
+    TX_SLEEP_TIME: float = 1.0
```
Comment (yakkle) [inline didsdk/config.py:8]: huh?
```suggestion
    # Second
    TX_SLEEP_TIME: int | float = 1.0
```

Comment (hyerim-kim) [reply didsdk/config.py]: The `|` operator does not work on version 3.8 even though imports the package. So, I use the keyword of typing `Union`.

File: didsdk/config.py (hunk 7 above)
Comment (hyerim-kim) [reply didsdk/config.py]: I applied that configuration can cover all of float and int.

**Review by yakkle — APPROVED** [review]

File: README.md [hunk 11]
```diff
@@ -2,3 +2,10 @@
 
 This SDK is used not only to create and manage `ICON DID`, but also to issue and verify `credentials` and `presentations`.
 
+### Configure
+Add some configurations of `didsdk` to a `.env` file in your project.
+~~~
+TX_RETRY_COUNT=int[default:5]
+TX_SLEEP_TIME=int[default:2]
```
Comment (yakkle) [inline README.md:9]: Need to update.

File: pyproject.toml [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+dev = [
+  "pytest~=7.2.2",
+  "pytest-asyncio==0.21.1",
+  "pytest-cov~=4.0.0",
+  "pytest-mock~=3.10.0",
+  "anyio[trio]~=3.7.0",
+]
+doc = [
+  "mkdocs-material~=9.1.18",
+]
```
Comment (yakkle) [inline pyproject.toml:44]: for future support?

Comment (hyerim-kim) [reply pyproject.toml]: 😅 removed !

File: README.md (hunk 11 above)
Comment (hyerim-kim) [reply README.md]: updated!

**Review by ddok2 — APPROVED** [review]

**Pull request merged**